### PR TITLE
Don't panic on empty file/buffer

### PR DESCRIPTION
**Pull request by notoria**

### CODE REVIEW

**Comment by notoria** [comment]
This pull request should solve #107 .

File: helix-core/src/match_brackets.rs [hunk 1]
```diff
@@ -28,6 +28,11 @@ pub fn find(syntax: &Syntax, doc: &Rope, pos: usize) -> Option<usize> {
     let end_byte = node.end_byte() - 1; // it's end exclusive
     let start_char = doc.byte_to_char(start_byte);
     let end_char = doc.byte_to_char(end_byte);
+    let len = doc.len_chars();
+
+    if start_char >= len || end_char >= len {
```
Comment (archseer) [inline helix-core/src/match_brackets.rs:33]: I think `start_char` and `end_char` should always be valid, we got them straight from `byte_to_char`. If they were out of bounds, then `byte_to_char` would panic.

Comment (notoria) [reply helix-core/src/match_brackets.rs:33]: I cannot reproduce #107 but the command 16xd should panic.

**Comment by notoria** [comment]
After further investigation, it's likely unrelated to #107. The issue is with empty file/buffer.

**Comment by kevinsjoberg** [comment]
@notoria this only solves the case when the file still contains a final newline. The following steps still causes `hx` to panic:

```console
touch a.txt
hx a.txt (and press any movement key)
```

**Comment by notoria** [comment]
I saw it but `match_brackets::find` isn't the culprit. It may be related with #112.

**Comment by pickfire** [comment]
Could be https://github.com/helix-editor/helix/pull/98 too

**Comment by kirawi** [comment]
> Could be related to #98 too

Doesn't fix it, the backtrace seems to indicate it's related to the LSP code.

**Comment by archseer** [comment]
Thanks! 🎉

**Pull request merged**